napari_btrack/sync.py

```diff
@@ -0,0 +1,117 @@
+"""
+This module contains functions for syncing widget values with TrackerConfig
+values.
+"""
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from magicgui.widgets import Container
+
+    from napari_btrack.config import UnscaledTrackerConfig
+
+
+def update_config_from_widgets(
+    unscaled_config: UnscaledTrackerConfig,
+    container: Container,
+):
+    """Update an UnscaledTrackerConfig with the current widget values."""
+
+    sigmas = unscaled_config.sigmas
+    sigmas.P = container.P_sigma.value
+    sigmas.G = container.G_sigma.value
+    sigmas.R = container.R_sigma.value
+
+    config = unscaled_config.tracker_config
+    config.update_method = (
+        container.update_method_selector._widget._qwidget.currentIndex()
```

Comment:
I'd rather not too but it's necessary. `magicgui` exposes the *name* of the update method selected, but not its *index* in the list of all possible methods. And `btrack` requires an enum, either 0 or 1, for selecting the update method, rather than the name of the method. So I'm directly accessing the underlying `QtWidget` and using the `currentIndex` method from that.

I guess we could do something like:
```python
update_method_name = container.update_method_selector.current_choice
update_method_index = container.update_method_selector.choices.index(update_method_name)
config.update_method - update_method_index
```
but it's a lot more verbose

Reply:
While it is more verbose, we have the guarantee it will work, right? We have no guarantee the above will work in the future. We could always add those three lines as a helper function?

Reply:
yeah okay, let's use this other approach for now then

Reply:
I've updated this to not use the private attributes of the widgets